azure/azure.go

```diff
@@ -114,6 +241,87 @@ func (a *azureOps) InstanceID() string {
 	return a.instance
 }
 
+func (a *azureOps) InspectInstance(instanceID string) (*cloudops.InstanceInfo, error) {
+
+	instInfo := &cloudops.InstanceInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			Name:   a.instance,
+			ID:     a.instance,
+			Zone:   "",
+			Region: "",
+		},
+	}
+	return instInfo, nil
+}
+
+func (a *azureOps) InspectInstanceGroupForInstance(instanceID string) (*cloudops.InstanceGroupInfo, error) {
+
+	ctx := context.Background()
+	agentPool, err := a.agentPoolsClient.Get(ctx, a.resourceGroupName, a.managedClusterName, a.agentPoolName)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get node pool details for [%s] of cluster [%v] in [%v] resource group."+
+			"Error: [%v] ", a.agentPoolName, a.managedClusterName, a.resourceGroupName, err)
+	}
+
+	// TODO: Currently multi-availability feature is in preview and
+	// zone field is not populated by azure agentPoolClient.
+	// For now, populate it with dummy value.
+	retval := &cloudops.InstanceGroupInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			ID:   *agentPool.ID,
+			Name: *agentPool.Name,
+		},
+		Zones: []string{"dummyzone"},
+	}
+
+	return retval, nil
+}
+
+func (a *azureOps) GetInstanceGroupSize(instanceGroupID string) (int64, error) {
+
+	ctx := context.Background()
+	agentPool, err := a.agentPoolsClient.Get(ctx, a.resourceGroupName, a.managedClusterName, instanceGroupID)
+	if err != nil {
+		return 0, err
+	}
+	return int64(*agentPool.Count), nil
+}
+
+// SetInstanceGroupSize sets desired node count per availability zone
+// for given instance group
+func (a *azureOps) SetInstanceGroupSize(instanceGroupID string,
+	count int64,
+	timeout time.Duration) error {
+
+	ctx := context.Background()
+	var cancel context.CancelFunc
+	if timeout > time.Nanosecond {
+		// If timeout is non-zero then
```

Comment:
*If timeout is zero then ...

Reply:
If timeout is zero then we want `SetInstanceGroupSize()` API to wait infinitely. If user wants API to be timebound then he needs to set non-zero timeout. 
If timeout is zero then no deadline is set to context `ctx` and WaitForCompletionRef() will not terminate on deadline.

Reply:
Yes I understood that, just change non-zero -> zero in the comment.

Reply:
Updated comment for both the cases.